src/python/pants/java/nailgun_io.py

```diff
@@ -16,21 +16,46 @@
 from pants.java.nailgun_protocol import ChunkType, NailgunProtocol
 
 
-@contextmanager
-def _pipe(isatty):
-  r_fd, w_fd = os.openpty() if isatty else os.pipe()
-  yield (r_fd, w_fd)
+class Pipe(object):
```

Comment:
Worth declaring as a datatype?

Reply:
Not for now, as we need default values for `self.writable`. Have left a TODO with the the link to a PR that will enable that.